packages/core/src/components/select/select.tsx

```diff
@@ -48,10 +50,60 @@ export class AtomSelect {
   @Event() atomDismiss!: EventEmitter<void>
   @Event() atomFocus!: EventEmitter<void>
 
+  @Method()
+  setBadge() {
+    const ionItemElements = document.querySelectorAll('ion-item')
+    const optionsWithBadge = this.optionsWithBadge
+
+    if (!ionItemElements || !optionsWithBadge) return
+
+    ionItemElements?.forEach((itemElement) => {
+      const elementText = itemElement.textContent?.trim()
+      const optionWithBadge = optionsWithBadge[elementText]
```

Comment:
Couldn't we add a validation on `this.options.map` instead? Like: 

```TS
{option.badge && (
                <atom-badge
                  class='atom-badge'
                  type={option.badge.type}
                  label={option.badge.label}
                ></atom-badge>
              )}
```

Reply:
No. The Shadow DOM encapsulates the component content, and Ionic renders the options content as plain text inside a native <select> element. This means that any custom styling or components are lost. As a workaround, we follow the approach of manipulating the DOM after rendering to add custom elements like badges.

Reply:
Jesus... What strange! It converts the ion-select-item to ion-item
![image](https://github.com/user-attachments/assets/4464fa56-ac34-4dc1-98e3-dcf60d4ea531)

I was seeing other alternatives to avoid/reduce this logic. Such as using `data-attribute`, but it looks like ion-select-item only[ re-pass the class attribute ](https://github.com/ionic-team/ionic-framework/blob/main/core/src/components/select-option/select-option.tsx#L27), so it (data-attribute) is also lost. 

Could you please add a comment to explain it?

Reply:
Done here https://github.com/juntossomosmais/atomium/pull/549#:~:text=and%20style%20space-,7611711,-feat%3A%20add%20example